### PR TITLE
Initial changes for refactoring auto renewal

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Part of #3822 

Initial PR for refactoring `auto-renewal` changes
- Added `RenewalWork`, `RemovalWork`, `ClassificationWork` to separate renewal, removal, classification functions

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiryModule.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.hedera.services.state.expiry.renewal.RenewalWork;
+import dagger.Module;
+import dagger.Provides;
+import javax.inject.Singleton;
+
+@Module(includes = {PropertiesModule.class})
+public interface ExpiryModule {
+    @Provides
+    @Singleton
+    static RenewalWork providesRenewalWork(
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiryModule.java:38]: Any reason to not just use an `@Inject` constructor for these bindings?

File: hedera-node/src/main/java/com/hedera/services/state/expiry/classification/ClassificationResult.java [hunk 2]
```diff
@@ -13,18 +13,16 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.hedera.services.state.expiry.renewal;
+package com.hedera.services.state.expiry.classification;
 
-public enum RenewableEntityType {
+public enum ClassificationResult {
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/state/expiry/classification/ClassificationResult.java:18]: Nice.

**Review by tinker-michaelj — DISMISSED** [review]
I think spotless needs a re-run, but looks great!!

File: hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiryModule.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiryModule.java]: Fixed as suggested using `@Binds` annotation. Thanks Michael

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Pull request merged**